modules/lang-painless/src/main/java/org/elasticsearch/painless/Def.java

```diff
@@ -72,14 +72,24 @@
 
         private static final MethodHandle OBJECT_ARRAY_MH = ARRAY_TYPE_MH_MAPPING.get(Object[].class);
 
+        // NOTE: the following are actually used, javac just does not know :)
+        @SuppressWarnings("unused")
```

Comment:
I have seen this, too (in Eclipse only). I would like to add the SuppressWarnings to the whole inner class holding that methods, not every method.
BTW: The PR for Java 9 for adding the missing getter was approved and was extended by a compiler instrinsic! I will think about using the Java 9 method to get the array length getter with a separate MH to get the arraylength MH.